cg/meta/qc_metrics/collect_qc_metrics.py

```diff
@@ -0,0 +1,85 @@
+"""Module for the collect qc metrics meta api."""
+
+import logging
+from typing import Iterator
+
+from housekeeper.store.models import File, Tag
+
+from cg.clients.arnold.api import ArnoldAPIClient
+from cg.clients.janus.api import JanusAPIClient
+from cg.clients.janus.dto.create_qc_metrics_request import (
+    CreateQCMetricsRequest,
+    FilePathAndTag,
+)
+from cg.clients.janus.exceptions import JanusClientError, JanusServerError
+from cg.exc import HousekeeperFileMissingError
+from cg.meta.meta import MetaAPI
+from cg.models.cg_config import CGConfig
+from cg.store.models import Case, Sample
+
+LOG = logging.getLogger(__name__)
+
+
+class CollectQCMetricsAPI(MetaAPI):
+    """Collect qc metrics API."""
+
+    def __init__(self, config: CGConfig):
+        super().__init__(config)
+        self.janus_api: JanusAPIClient = config.janus_api
+        self.arnold_api: ArnoldAPIClient = config.arnold_api
+
+    def get_qc_metrics_file_paths_for_case(self, case_id: str) -> list[File]:
+        """Get the multiqc metrics files for a case."""
+        files: list[File] = self.housekeeper_api.get_files(
+            bundle=case_id, tags=["qc-metrics", "janus"]
+        ).all()
+        if not files:
+            raise HousekeeperFileMissingError(f"No qc metrics files found for {case_id}.")
+        return files
+
+    @staticmethod
+    def get_tag_to_include(tags: list[Tag]) -> str:
+        """Get the multiqc module file tag."""
+        tags_to_exclude: list[str] = ["qc-metrics", "janus", "multiqc"]
+        for tag in tags:
+            if tag.name not in tags_to_exclude:
+                return tag.name
+
+    def get_file_paths_and_tags(self, files: list[File]) -> list[FilePathAndTag]:
```

Comment:
"Why do we want to use tags and send them to Janus? Don't we only care about the paths there? It seems to me that this is just adding another layer of complexity and would become annoying to maintain if, for some reason, we have to change some tags. Wouldn't it be easier to just send a dictionary of files with descriptive keys? Maybe I'm missing something, but it seems to me we're adding some unnecessary logic.

Reply:
The tags are used in janus to retrieve the proper file model and parsing function.

For example somalier, fastp require a special parsing function. Using the tag I can make link the functions to the files. Alternative would be to rely on file names.